plugin/web/extensions/sidebar/core/components/content/common/DatasetCard/Field/Fields/3dtiles/BuildingColor/conditions.ts

```diff
@@ -0,0 +1,93 @@
+  `(${conditionalValue} >= ${num})`;
+
+const equalString = (prop: string, value: string) => `(\${${prop}} === "${value}")`;
+const equalNumber = (prop: string, value: number) => `(\${${prop}} === ${value})`;
+
+type Condition = [condition: string, result: string];
+
+const DEFAULT_CONDITION: Condition = ["true", "color('white')"];
+
+const conditionalHeight = defaultConditionalNumber("計測高さ");
+const HEIGHT_CONDITIONS: Condition[] = [
+  [compareGreaterThan(conditionalHeight, 180), "color('#F7FF00')"],
+  [compareRange(conditionalHeight, [120, 180]), "color('#FFCD00')"],
+  [compareRange(conditionalHeight, [60, 120]), "color('#F0D37B')"],
+  [compareRange(conditionalHeight, [31, 60]), "color('#E1CEE8')"],
+  [compareRange(conditionalHeight, [12, 31]), "color('#F7FF00')"],
+  [compareRange(conditionalHeight, [0, 12]), "color('#382A54')"],
+  DEFAULT_CONDITION,
+];
+
+const conditionalPurpose = "用途";
+const PURPOSE_CONDITIONS: Condition[] = [
+  [equalString(conditionalPurpose, "業務施設"), "color('#ff7f50')"],
+  [equalString(conditionalPurpose, "商業施設"), "color('#ff4500')"],
+  [equalString(conditionalPurpose, "宿泊施設"), "color('#ffff00')"],
+  [equalString(conditionalPurpose, "商業系複合施設"), "color('#ff4500')"],
+  [equalString(conditionalPurpose, "住宅"), "color('#32cd32')"],
+  [equalString(conditionalPurpose, "共同住宅"), "color('#00ff7f')"],
+  [equalString(conditionalPurpose, "店舗等併用住宅"), "color('#00ffff')"],
+  [equalString(conditionalPurpose, "店舗等併用共同住宅"), "color('#00ffff')"],
+  [equalString(conditionalPurpose, "作業所併用住宅"), "color('#00ffff')"],
+  [equalString(conditionalPurpose, "官公庁施設"), "color('#4169e1')"],
+  [equalString(conditionalPurpose, "文教厚生施設"), "color('#0000ff')"],
+  [equalString(conditionalPurpose, "運輸倉庫施設"), "color('#9370db')"],
+  [equalString(conditionalPurpose, "工場"), "color('#87cefa')"],
+  [equalString(conditionalPurpose, "農林漁業用施設"), "color('#008000')"],
+  [equalString(conditionalPurpose, "併給処理施設"), "color('#8b4513')"],
+  [equalString(conditionalPurpose, "防衛施設"), "color('#b22222')"],
+  [equalString(conditionalPurpose, "その他"), "color('#d8bfd8')"],
+  [equalString(conditionalPurpose, "不明"), "color('#e6e6fa')"],
+  DEFAULT_CONDITION,
+];
+
+const conditionalStructure = "建物構造";
+const STRUCTURE_CONDITIONS: Condition[] = [
+  [equalString(conditionalStructure, "耐火構造"), "color('#7c7b87')"],
+  [equalString(conditionalStructure, "防火造"), "color('#bc8f8f')"],
+  [equalString(conditionalStructure, "準防火造"), "color('#d6caae')"],
+  [equalString(conditionalStructure, "木造"), "color('#d2b48c')"],
+  DEFAULT_CONDITION,
+];
+
+const conditionalFireproof = "建物利用現況_耐火構造種別";
+const FIREPROOF_CONDITIONS: Condition[] = [
+  [equalString(conditionalFireproof, "耐火"), "color('#7c7b87')"],
+  [equalString(conditionalFireproof, "準耐火造"), "color('#8e9bb0')"],
+  [equalString(conditionalFireproof, "その他"), "color('#d083af')"],
+  [equalString(conditionalFireproof, "不明"), "color('#89c2ea')"],
+  DEFAULT_CONDITION,
+];
```

Comment:
Are these hardcoded values what we want? (Did @rot1024 say to do it like this?)

Reply:
I discussed with @lavalse. I decided we need to do this for now,  but we should have more efficient way.

Reply:
Yes. I agree we need a more efficient way. But I think it is ok now to do this.